gooddata-sdk/gooddata_sdk/catalog/data_source/declarative_model/physical_model/column.py

```diff
@@ -0,0 +1,51 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any, Optional
+
+from gooddata_metadata_client.model.declarative_column import DeclarativeColumn
+
+
+class CatalogDeclarativeColumn:
+    def __init__(
+        self,
+        name: str,
+        data_type: str,
+        is_primary_key: bool,
+        referenced_table_id: Optional[str],
+        referenced_table_column: Optional[str],
+    ):
+        self.name = name
+        self.data_type = data_type
+        self.is_primary_key = is_primary_key
+        self.referenced_table_id = referenced_table_id
+        self.referenced_table_column = referenced_table_column
+
+    @classmethod
+    def from_api(cls, entity: dict[str, Any]) -> CatalogDeclarativeColumn:
+        return cls(
+            name=entity["name"],
+            data_type=entity["data_type"],
+            is_primary_key=entity["is_primary_key"],
+            referenced_table_id=entity.get("referenced_table_id"),
+            referenced_table_column=entity.get("referenced_table_column"),
+        )
+
+    def to_api(self) -> DeclarativeColumn:
+        kwargs: dict[str, Any] = dict()
+        if self.referenced_table_id:
+            kwargs["referenced_table_id"] = self.referenced_table_id
+        if self.referenced_table_column:
+            kwargs["referenced_table_column"] = self.referenced_table_column
+        return DeclarativeColumn(name=self.name, data_type=self.data_type, is_primary_key=self.is_primary_key, **kwargs)
```

Comment:
is_primary_key is optional. move it to kwargs

Reply:
fixed